packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -4146,42 +4570,77 @@ class ObjectsInRedisClient {
         });
     }
 
-    extendObjectAsync(id, obj, options) {
+    extendObjectAsync(
+        id: string,
+        obj: Partial<ioBroker.Object>,
+        options: ioBroker.ExtendObjectOptions
+    ): Promise<ioBroker.CallbackReturnTypeOf<ioBroker.ExtendObjectCallback>> {
         return new Promise((resolve, reject) =>
             this.extendObject(id, obj, options, (err, res) => (err ? reject(err) : resolve(res)))
         );
     }
 
-    setConfig(id, obj, options, callback) {
+    setConfig(
+        id: string,
+        obj: ioBroker.SettableObject,
+        options: CallOptions,
+        callback: ioBroker.SetObjectCallback
+    ): void | Promise<ioBroker.CallbackReturnTypeOf<ioBroker.SetObjectCallback>> {
```

Comment:
setConfig most likely accepts a `SettableOtherObject & {type: "config"}`, right?
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/iobroker/objects.d.ts#L717

Reply:
From a syntax view it is an alias for `setObject` but probably it is meant to be what you have stated. @Apollon77 maybe knows it.